src/windows/View.tsx

```diff
@@ -0,0 +1,371 @@
+/**
+* View.tsx
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* Windows-specific implementation of View.
+*/
+
+import React = require('react');
+import RNW = require('react-native-windows');
+import Types = require('../common/Types');
+import PropTypes = require('prop-types');
+
+import {View as ViewCommon} from '../native-common/View';
+import EventHelpers from '../native-common/utils/EventHelpers';
+import { applyFocusableComponentMixin, FocusManagerFocusableComponent, FocusManager } from '../native-desktop/utils/FocusManager';
+
+const KEY_CODE_ENTER = 13;
+const KEY_CODE_SPACE = 32;
+
+const DOWN_KEYCODES = [KEY_CODE_SPACE, KEY_CODE_ENTER];
+const UP_KEYCODES = [KEY_CODE_SPACE];
+
+export interface ViewContext {
+    isRxParentAText?: boolean;
+    focusManager?: FocusManager;
+}
+
+// Simple check for the presence of the updated React Native for Windows
+const HasFocusableWindows = (RNW.FocusableWindows !== undefined);
+
+export class View extends ViewCommon implements React.ChildContextProvider<ViewContext>, FocusManagerFocusableComponent {
+    static contextTypes: React.ValidationMap<any> = {
+        isRxParentAText: PropTypes.bool,
+        focusManager: PropTypes.object
+    };
+    context: ViewContext;
+
+    static childContextTypes: React.ValidationMap<any> = {
+        isRxParentAText: PropTypes.bool.isRequired,
+        focusManager: PropTypes.object
+    };
+
+    private _onKeyDown: (e: React.SyntheticEvent<any>) => void;
+    private _onMouseEnter: (e: React.SyntheticEvent<any>) => void;
+    private _onMouseLeave: (e: React.SyntheticEvent<any>) => void;
+    private _onMouseOver: (e: React.SyntheticEvent<any>) => void;
+    private _onMouseMove: (e: React.SyntheticEvent<any>) => void;
+
+    private _focusableElement : RNW.FocusableWindows | null = null;
+
+    private _focusManager: FocusManager;
+    private _isFocusLimited: boolean;
+
+    constructor(props: Types.ViewProps, context: ViewContext) {
+        super(props);
+
+        if (props.restrictFocusWithin || props.limitFocusWithin) {
+            this._focusManager = new FocusManager(context && context.focusManager);
+
+            if (props.limitFocusWithin) {
+                this.setFocusLimited(true);
+            }
+        }
+    }
+
+    componentWillReceiveProps(nextProps: Types.ViewProps) {
+        super.componentWillReceiveProps(nextProps);
+
+        if (!!this.props.restrictFocusWithin !== !!nextProps.restrictFocusWithin) {
+            console.error('View: restrictFocusWithin is readonly and changing it during the component life cycle has no effect');
+        } else if (!!this.props.limitFocusWithin !== !!nextProps.limitFocusWithin) {
```

Comment:
>limitFocusWithin [](start = 32, length = 16)

Are `restrictFocusWithin` and `limitFocusWithin` mutually exclusive? If not, we should use a second `if` rather than an `else if`

Reply:
True, copied from elsewhere. I'll change both places.

---
In reply to: [159359222](https://github.com/Microsoft/reactxp/pull/404#discussion_r159359222) [](ancestors = 159359222)